### PR TITLE
add support to PythonPackage to fix 'python' shebang lines in specified files to '#!/usr/bin/env python'

### PR DESCRIPTION
**Pull request by boegel**
edit: now requires ~~https://github.com/easybuilders/easybuild-framework/pull/2844~~

### CODE REVIEW

File: easybuild/easyblocks/generic/pythonpackage.py [hunk 1]
```diff
@@ -558,6 +560,15 @@ def install_step(self):
         if pythonpath is not None:
             env.setvar('PYTHONPATH', pythonpath, verbose=False)
 
+        for glob_pattern in self.cfg.get('fix_python_shebang_for', []):
+            paths = glob.glob(os.path.join(self.installdir, glob_pattern))
+            self.log.info("Fixing shebang to '%s' for files that match '%s': %s", PYTHON_SHEBANG, glob_pattern, paths)
+            for path in paths:
+                apply_regex_substitutions(path, [(r'^#!.*python$', PYTHON_SHEBANG)])
```
Comment (akesandgren) [inline easybuild/easyblocks/generic/pythonpackage.py]: Need to be python[0-9.]*$ to cover all cases.

File: easybuild/easyblocks/generic/pythonpackage.py [hunk 2]
```diff
@@ -558,6 +560,15 @@ def install_step(self):
...
+
+                # remove backup that is made by apply_regex_substitutions
+                remove_file(path + '.orig.eb')
```
Comment (akesandgren) [inline easybuild/easyblocks/generic/pythonpackage.py]: Doesn't apply_regex have a flag to not create backup file?

File: easybuild/easyblocks/generic/pythonpackage.py (hunk 1 above)
Comment (akesandgren) [reply easybuild/easyblocks/generic/pythonpackage.py]: For python 3.7.2 it puts
#!/hpc2n/eb/software/Compiler/GCCcore/8.2.0/Python/3.7.2/bin/python3.7
in the file it created.

File: easybuild/easyblocks/generic/pythonpackage.py (hunk 2 above)
Comment (boegel) [reply easybuild/easyblocks/generic/pythonpackage.py]: No, it doesn't. We could add one though...

File: easybuild/easyblocks/generic/pythonpackage.py (hunk 1 above)
Comment (boegel) [reply easybuild/easyblocks/generic/pythonpackage.py]: Good point, thanks.

Comment (boegel) [reply easybuild/easyblocks/generic/pythonpackage.py]: Fixed

File: easybuild/easyblocks/generic/pythonpackage.py (hunk 2 above)
Comment (boegel) [reply easybuild/easyblocks/generic/pythonpackage.py]: @akesandgren support for skipping backup added in https://github.com/easybuilders/easybuild-framework/pull/2844, and started using it in cc1464a31

**Comment by akesandgren** [comment]
Going in, thanks @boegel!

**Pull request merged**